lib/core/src/Cardano/Wallet/Api/Types/BlockHeader.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE RecordWildCards #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Cardano.Wallet.Api.Types.BlockHeader where
+
+import Prelude hiding
+    ( id )
+
+import Cardano.Wallet.Api.Helpers
+    ( ApiT (..), fromTextJSON, toTextJSON )
+import Cardano.Wallet.Primitive.Types
+    ( BlockHeader (..), SlotNo (..) )
+import Cardano.Wallet.Primitive.Types.Hash
+    ( Hash (..) )
+import Data.Aeson
+    ( FromJSON (parseJSON), ToJSON (toJSON) )
+import Data.Quantity
+    ( Quantity (..) )
+import Data.Word
+    ( Word32, Word64 )
+import GHC.Generics
+    ( Generic )
+
+instance FromJSON (ApiT (Hash "BlockHeader")) where
+    parseJSON = fromTextJSON "Block Header Hash"
+
+instance ToJSON (ApiT (Hash "BlockHeader")) where
+    toJSON = toTextJSON
+
+data ApiBlockHeader = ApiBlockHeader
+  { id :: ApiT (Hash "BlockHeader"),
+    slot :: Quantity "slot" Word64,
+    height :: Quantity "block" Word32
+  }
+  deriving (Eq, Show, Generic, FromJSON, ToJSON)
+
+mkApiBlockHeader :: BlockHeader -> ApiBlockHeader
+mkApiBlockHeader BlockHeader {..} = ApiBlockHeader {..}
+  where
+    id = ApiT headerHash
+    height = blockHeight
+    slot = Quantity $ unSlotNo slotNo
```

Comment:
I am not a fan of this approach to create `ToJSON` instances because it couples field name in Haskell to a field name in JSON. 
I'd either have a manual instance (non derived one) or use a `fieldLabelModifier` that lookups fields in a translation table.

Reply:
So @HeinrichApfelmus suggested we do not repeat prefixes in field names of records because we decided to use #id as selectors that disambiguate in case of records with the same field names. I have no opinions, but there has to be one idea to follow, if I have to follow it

Reply:
Note, I wasn't talking about prefixes, I suggested to decouple field names from JSON object keys (with or without prefixes)

Reply:
I do not see the advantage of the decoupling here as the names make sense to me. Any better naming ?

Reply:
There are 2 sets of attribute names: 

The ones that are in the `BlockHeader` type:
- `slotNo`
- `blockHeight`
- `headerHash`
- `parentHeaderHash`

The new ones you propose for the `ApiBlockHeader`
- `id`
- `slot`
- `height`

You say that 2nd set makes sense to you. Does it mean that the first set doesn't make sense?
Please help me to understand why should we maintain 2 different name sets?
The reason not to expose the `parentHeaderHash` is also not clear to me.

Reply:
it seems to me `parentHeaderHash` is not computable from `currentTip`

Reply:
the only reason to change them is to make them shorter in the API. I will use the old one  with camel2underscore conversion instead

```
slot_no
block_height
header_hash
```

make sense ?